postcode_lookup.R

```diff
@@ -0,0 +1,149 @@
+#####################################################
+# Postcode Lookup
+# Author: Catherine Holland
+# Date: February 2022
+# Written on RStudio Server
+# Version of R - 3.6.1
+# Input -
+# Description -
+#####################################################
+
+library(dplyr)
+library(tidyr)
+library(haven)
+
+
+## read data in ##
+
+# postcode data
+pc_file <- read_spd_file()
+
+# simd data
+read_simd_file <- function(file) {
+  simd_path <- fs::path(get_lookups_dir(), "Deprivation", file)
+  # If given a sav extension (or other), swap it for rds
+  simd_path <- fs::path_ext_set(simd_path, "rds")
+  # Check if the file exists and we can read it
+  if (!fs::file_access(simd_path, "read")) {
+    rlang::abort(message = "Couldn't read the simd file")
+  }
+  return(simd_path)
+}
+simd_file <- readr::read_rds(read_simd_file(file = "postcode_2021_2_simd2020v2.rds"))
+
+# locality
+read_locality_file <- function(file) {
+  locality_path <-
+    fs::path(get_lookups_dir(), "Geography", "HSCP Locality", file)
+  # If given a sav extension (or other), swap it for rds
+  locality_path <- fs::path_ext_set(locality_path, "rds")
+  # Check if the file exists and we can read it
+  if (!fs::file_access(locality_path, "read")) {
+    rlang::abort(message = "Couldn't read the locality file")
+  }
+  return(locality_path)
+}
+locality_file <- readr::read_rds(read_locality_file("HSCP Localities_DZ11_Lookup_20200825.rds"))
+
+
+## clean up data ##
+
+# arrange pc lookups based on pc7
+pc_file <-
+  pc_file %>%
+  arrange(pc7)
+
+# arrange simd based on pc7
+simd_file <-
+  simd_file %>%
+  arrange(pc7)
+
+# join data together by pc7
+data <-
+  pc_file %>%
+  left_join(simd_file, by = "pc7") %>%
+  # sort by DataZone2011
+  rename(datazone2011 = "datazone2011.x") %>%
+  arrange(datazone2011)
```

Comment:
Assumedly this is because both files have a datazone2011 variable. I think a better way would be to have dropped it from one of the files earlier (probably makes the most sense to drop it from the SIMD file). Alternatively, you could match by pc7 and datazone2011 (they will be 1:1 so will do the same thing but not end up duplicating the column).
```suggestion
data <-
  pc_file %>%
  left_join(simd_file, by = c("pc7", "datazone2011")) 
```